### PR TITLE
Feature/local feature flag

### PR DESCRIPTION
**Pull request by mohammadkhizerkhan**
- adding local feature flag 
- it will switch from prod to stag and vice-versa
Screenshot 2023-09-13 at 11 51 03 PM
Screenshot 2023-09-13 at 11 52 53 PM
Screenshot 2023-09-13 at 11 53 20 PM

https://github.com/Real-Dev-Squad/mobile-app/assets/64767673/bf0a3426-ddb1-44ac-9829-2a15e976e032

### CODE REVIEW

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 1]
```diff
@@ -257,6 +261,18 @@ const AuthScreen = () => {
           text={Strings.SIGN_IN_WITH_WEB}
           onPress={activateCamera}
         />
+        <AuthScreenButton
+          text={
+            API_BASE_URL === 'https://api.realdevsquad.com/'
```
Comment (shreya-mishra) [inline src/screens/AuthScreen/AuthScreen.tsx]: can we check this in some other file?

File: src/reducers/featureFlag.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+export const featureFlagState = {
+  API_BASE_URL: 'https://api.realdevsquad.com/',
```
Comment (shreya-mishra) [inline src/reducers/featureFlag.reducer.ts]: let's move base api url in constant file

File: src/reducers/featureFlag.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+const localFeatureFlag = (state = featureFlagState, action: any) => {
+  switch (action.type) {
+    case 'STAGING':
+      return {
+        ...state,
+        API_BASE_URL: 'https://staging-api.realdevsquad.com/',
```
Comment (shreya-mishra) [inline src/reducers/featureFlag.reducer.ts]: doesnt look good

**Review by shreya-mishra — COMMENTED** [review]
please fix below comments

**Comment by sahsisunny** [comment]
@mohammadkhizerkhan, should we hide the feature flag or make it less visible to normal users? If so, why give them a button to switch between different environments? We could add gestures or toggle buttons to switch instead.

**Comment by mohammadkhizerkhan** [comment]
> @mohammadkhizerkhan, should we hide the feature flag or make it less visible to normal users? If so, why give them a button to switch between different environments? We could add gestures or toggle buttons to switch instead.

yes, i need to discuss this with @shreya-mishra.
and i think we are not going to show this feature to normal user

**Review by shreya-mishra — COMMENTED** [review]
can I have the video of switching the feature as well for example if we are in prod then it should show only profile and home screen otherwise it should show all three tabs

**Pull request closed without merging**